src/timeout.c

```diff
@@ -175,7 +177,7 @@ int getTimeoutFromObjectOrReply(client *c, robj *object, mstime_t *timeout, int
             addReplyError(c, "timeout is out of range");
             return C_ERR;
         }
-        tval = (long long) ftval;
+        tval = (long long) ceil(ftval);
```

Comment:
I think we should use `ceill()` instead of `ceil()` here. `ftval` is `long double`.